leadership.go

```diff
@@ -128,6 +132,7 @@ func (l *Lock) RefreshLock(ctx context.Context) error {
 
 	l.refreshFailures = 0
 	l.latestMetadataGeneration = attrs.Metageneration
+	l.latestGeneration = attrs.Generation
```

Comment:
I think the behaviour here is that `l.latestGeneration` should always be `0`. If it's something else, it means that someone updated the lock, which should happen.

Reply:
My suggestion was incorrect, so resolving the comment